### ISSUE TITLE
Checkout confirmation appears multiple times after purchase in `/start` flow

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#50432]
When creating new site and purchasing a new plan, I see confirmation multiple times when landing at Customer home page:

<img src="https://user-images.githubusercontent.com/87168/109001919-36c1f500-76ae-11eb-97fb-b97ed00c81d7.png" width="300"/>

https://user-images.githubusercontent.com/87168/109001990-4d684c00-76ae-11eb-8bad-7d3365309546.mov

cc @Automattic/shilling  is this something you'd be able to have a look into?

**Comment by michaeldcain** [comment]
@simison I'm having trouble reproducing this. How did you go from a logged-out plans page to a logged-in Calypso instance?

**Comment by sirbrillig** [comment]
I believe that message has nothing to do with Checkout and comes from the Customer Home page here: 

https://github.com/Automattic/wp-calypso/blob/bb4367f11d9e63ccd62a755fcc1f654df4bb1f80/client/my-sites/customer-home/main.jsx#L65-L69

That hook displays that message on every render where one of the other props changes as long as `noticeType` is set (although it's not clear to me what's changing it must be either `layout` or `canUserUseCustomerHome`). I'm thinking it might be `layout` since that's an object, but it's not clear to me where that prop comes from. 

The notice was added in https://github.com/Automattic/wp-calypso/pull/49612 (cc @niranjan-uma-shankar) and I think the logic might be missing something to say "if we've already displayed this message, don't display it again" or possibly just a way to make sure none of the other dependencies changes unexpectedly; for example we could remove all the dependencies except for `noticeType` and tell the linter to ignore it. (The additions from that PR just highlighted the problem. It looks like it's been like that since https://github.com/Automattic/wp-calypso/pull/49301).
